azurerm/internal/tools/website-scaffold/main.go

```diff
@@ -0,0 +1,891 @@
+	}
+	if name == "location" {
+		return "\"West Europe\""
+	}
+	if name == "resource_group_name" {
+		return "\"example-resources\""
+	}
+
+	return "\"TODO\""
+}
+
+func (gen documentationGenerator) distinctBlockNames(input []string) []string {
+	// this is a delightful hack to work around multiple blocks being a thing
+	temp := make(map[string]struct{}, 0)
+	for _, v := range input {
+		temp[v] = struct{}{}
+	}
+
+	output := make([]string, 0)
+	for k := range temp {
+		output = append(output, k)
+	}
+
+	return output
+}
+
+func (gen documentationGenerator) processElementForExample(field string, indentLevel int, elem interface{}, isAttribute bool) string {
+	indent := gen.buildIndentForExample(indentLevel)
+
+	// it's an array of something, work out what
+	if array, ok := elem.(*schema.Schema); ok {
+		switch array.Type {
+		case schema.TypeString:
+			return fmt.Sprintf("%s%s = [ \"example\" ]\n", indent, field)
+
+		case schema.TypeInt:
+			return fmt.Sprintf("%s%s = [ 1 ]\n", indent, field)
+
+		default:
+			return "TODO"
+		}
+	}
+
+	// otherwise it's a list so we're gonna have to go around
+	if list, ok := elem.(*schema.Resource); ok && len(list.Schema) > 0 {
+		innerFields := gen.requiredFieldsForExampleBlock(list.Schema, indentLevel+1)
+		attributeSyntax := ""
+		if isAttribute {
+			attributeSyntax = " ="
+		}
+		return fmt.Sprintf("\n%s%s%s {\n%s  %s\n%s}\n", indent, field, attributeSyntax, innerFields, indent, indent)
+	}
+
+	// unless something's broken, since this is likely during provider dev it's likely things could be missing
+	panic("Field %q has an Element but isn't a Set or List - double-check the schema")
+}
+
+func (gen documentationGenerator) requiredFieldsForExampleBlock(fields map[string]*schema.Schema, indentLevel int) string {
+	indent := gen.buildIndentForExample(indentLevel)
+	output := ""
+
+	var processField = func(name string, field *schema.Schema) string {
+		value := gen.determineDefaultValueForExample(name, field)
+		return fmt.Sprintf("%s%s = %s\n", indent, name, value)
+	}
+
+	// if we have a "name", "location" "resource_group_name" field output those first as per convention
+	if v, ok := fields["name"]; ok && v.Required {
+		output += processField("name", v)
+	}
+	if v, ok := fields["resource_group_name"]; ok && v.Required {
+		output += processField("resource_group_name", v)
+	}
+	if v, ok := fields["location"]; ok && v.Required {
+		output += processField("location", v)
+	}
+
+	for field, v := range fields {
+		if !v.Required {
+			continue
+		}
+		if field == "location" || field == "name" || field == "resource_group_name" {
+			continue
+		}
+
+		if v.Elem != nil {
+			isAttribute := v.ConfigMode == schema.SchemaConfigModeAttr
+			output += gen.processElementForExample(field, indentLevel, v.Elem, isAttribute)
+			continue
+		}
+
+		output += processField(field, v)
+	}
+	return strings.TrimSuffix(output, "\n")
+}
+
+func (gen documentationGenerator) sortFields(input map[string]*schema.Schema) []string {
```

Comment:
Alphabetic ordering might not obey the conventions that:

1. Some special fields have special ordering, e.g. `name` comes first and `tags` comes last
2. Except point 1, `ForceNew` + `Required` > `ForceNew` + `Optional` > `Required` > `Optional`